### PR TITLE
live-tests: add duckdb backend

### PR DESCRIPTION
**Pull request by alafanechere**
Small 🎁 for @bleonard : the live testing tool persists airbyte messages to a DuckDb file.

### CODE REVIEW

**Comment by bleonard** [comment]
It's not a present for me. This will help run comparisons at scale!

**Review by erohmensing — APPROVED** [review]

**Comment by bleonard** [comment]
I bet @aaronsteers knows a way to load into duck directly from the jsonl files

**Comment by aaronsteers** [comment]
Indeed! There's a function called read_json_auto(), which we're using in PyAirbyte.


https://duckdb.org/docs/data/json/overview.html#read_json_auto-function

File: airbyte-ci/connectors/live-tests/src/live_tests/commons/backends/duckdb_backend.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            if _message.type is AirbyteMessageType.CONNECTION_STATUS:
+                messages_by_type.setdefault(_message.type, []).append(json.loads(_message.connectionStatus.json()))
+            if _message.type is AirbyteMessageType.SPEC:
+                messages_by_type.setdefault(_message.type, []).append(json.loads(_message.spec.json()))
+            if _message.type is AirbyteMessageType.TRACE:
+                messages_by_type.setdefault(_message.type, []).append(json.loads(_message.trace.json()))
+            if _message.type is AirbyteMessageType.LOG:
+                messages_by_type.setdefault(_message.type, []).append(json.loads(_message.log.json()))
+            if _message.type is AirbyteMessageType.CONTROL:
+                messages_by_type.setdefault(_message.type, []).append(json.loads(_message.control.json()))
```
[the remark this answers is not in the document]
Comment (alafanechere) [reply airbyte-ci/connectors/live-tests/src/live_tests/commons/backends/duckdb_backend.py:38]: Oh yeah, good tips, thank you!

**Comment by alafanechere** [comment]
@bleonard @aaronsteers @erohmensing thanks for your suggestion
I ended up re-using the `FileBackend` logic to write `jsonl` files and then load them to Duckdb.
I think it has a reduced memory footprint compared to the use of Pandas. And it makes the backend code a lot lighter.

**Pull request merged**